### PR TITLE
Fix bug where we were not sending topics to the correct content items

### PR DESCRIPTION
**Pull request by AndrewVos**
~~Don't merge, because this needs tests and need to fix the bug in the `publishing-api` if possible.~~
Also, turns out if you pass an incorrect ID through to `patch_links`,
you get a 200.

### CODE REVIEW

**Comment by thehenster** [comment]
I'd be up for changing that `topic_id:` parameter too.. It should be `topic_content_id:` in the class and instance method?

I think it was me that introduced this bug but I'm blaming the naming :)

**Comment by AndrewVos** [comment]
Yeah, not sure why the tests didn't fail when I opened this PR :/

**Comment by AndrewVos** [comment]
Definitely not ready for merge. Need to fix the publishing API bug

**Comment by thehenster** [comment]
I think as it stands it fixes the problem no? My suggestion was just for readability.

**Comment by AndrewVos** [comment]
Added a test and refactored. Merge pls

**Comment by dougdroper** [comment]
:+1:

File: spec/models/guide_tagger_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          .with(
+            guide_content_id,
+            links: { topics: [topic.content_id] },
+          )
+      end
+
+      GuideTaggerJob.batch_perform_later(topic)
+    end
+  end
+end
```
Comment (thehenster) [inline spec/models/guide_tagger_job_spec.rb]: Good work adding this :clap:

Comment (AndrewVos) [inline spec/models/guide_tagger_job_spec.rb]: Thanks mate, I'll teach you about TDD someday ;)

Comment (AndrewVos) [inline spec/models/guide_tagger_job_spec.rb]: OOooh never mind I checked `git blame` and I'm gonna have to retract my burn. Nice work @tadast

File: spec/models/topic_spec.rb [hunk 2]
```diff
@@ -49,11 +49,13 @@
...
-      topic = create(:topic, tree: [{'guides' => ['2', '5']}, {'guides' => ['3']}])
-
-      expect(topic.guide_ids).to match_array([2, 3, 5])
+  describe "#guide_content_ids" do
+    it "returns the associated guide content ids" do
+      topic = create(:topic, :with_some_guides)
+      guide_ids = topic.tree.map {|t| t["guides"]}.flatten
+      guide_content_ids = Guide.where(id: guide_ids).pluck(:content_id)
+      expect(topic.guide_content_ids).to match_array(guide_content_ids)
+      expect(topic.guide_content_ids).to_not be_empty
```
Comment (thehenster) [inline spec/models/topic_spec.rb]: I find this less readable than the original but I'm not too sure how to improve it :( The content ids are generated so I guess we'll never get that nice readability of seeing the literals in the setup and the assertions.. Oh well.

Maybe stick a line break above the `expect`s? That'll fix it ;)

**Comment by thehenster** [comment]
Defo better passing the topic rather than the arguments for the publishing-api call.

File: spec/models/topic_spec.rb (hunk 2 above)
Comment (AndrewVos) [inline spec/models/topic_spec.rb]: I could set the `tree` attribute in the test I suppose. :(

**Pull request merged**